here-naksha-lib-handlers/src/main/java/com/here/naksha/lib/handlers/val/MockValidationHandler.java

```diff
@@ -0,0 +1,149 @@
+   *
+   * @param event the event to process.
+   * @return the result.
+   */
+  @Override
+  public @NotNull Result processEvent(@NotNull IEvent event) {
+    final NakshaContext ctx = NakshaContext.currentContext();
+    final Request<?> request = event.getRequest();
+
+    logger.info("Handler received request {}", request.getClass().getSimpleName());
+
+    return null;
+  }
+
+  protected @NotNull Result validateHandler(final @NotNull Request<?> request) {
+    if (!(request instanceof ContextWriteFeatures<?, ?, ?, ?, ?> cwf))
+      throw new XyzErrorException(
+          XyzError.NOT_IMPLEMENTED,
+          "Unsupported request type " + request.getClass().getSimpleName());
+
+    final @Nullable List<XyzFeature> violations = validateFeatures(cwf, cwf.getContext());
+    return HandlerUtil.createContextResultFromCodecList(cwf.features, cwf.getContext(), violations);
+  }
+
+  protected @Nullable List<XyzFeature> validateFeatures(
+      final @NotNull ContextWriteFeatures<?, ?, ?, ?, ?> cwf, final @Nullable List<?> context) {
+    // For random features from the input list, create 0-to-N random violations
+    final List<XyzFeature> violations;
+
+    // Decide randomly whether to generate violations or not
+    // Generation violations if odd number of features supplied, otherwise not
+    final boolean generateViolation = (cwf.features.size() % 2) > 0;
+
+    if (!generateViolation) return null;
+
+    // TODO : Write validation logic.
+
+    // Generate random violations and attach feature references
+    violations = new ArrayList<>();
+    int featureCnt = 0;
+    final List<XyzFeature> features = HandlerUtil.getXyzFeaturesFromCodecList(cwf.features);
+    for (final XyzFeature feature : features) {
+      featureCnt++;
+      // Distribution of "count" of violations, depends on feature "number",
+      // using min condition (i.e. min (feature, violation count))
+      // For example, if we have 4 features and 3 mock violations, then:
+      //    feature #1, will have 1 violation i.e. min(1,3)
+      //    feature #2, will have 2 violations i.e. min(2,3)
+      //    feature #3, will have 3 violations i.e. min(3,3)
+      //    feature #4, will have 3 violations i.e. min(4,3)
+      int violationsCount = Math.min(featureCnt, totalViolations);
+      violations.addAll(getNViolationsWithFeatureReference(
+          violationsCount,
+          feature.getId(),
+          cwf.getCollectionId(),
+          feature.get("momType").toString()));
+    }
+    return violations;
+  }
+
+  private @NotNull List<XyzFeature> getNViolationsWithFeatureReference(
+      final int count,
+      final @NotNull String featureId,
+      final @NotNull String spaceId,
+      final @Nullable String featureType) {
+    final List<XyzFeature> violations = new ArrayList<>();
+    for (int i = 0; i < count && i < totalViolations; i++) {
+      final XyzFeature violation = parseJson(toJson(mockViolations.get(i)), XyzFeature.class);
```

Comment:
This made me think - since `Violation` model is already known, wouldn't we like it create the specific type for it?
We wouldn't have to add "custom" fields that are not really custom (like in a line below: `violation.put("violatedObject", reference)`)